Assets/Scripts/Player.cs

```diff
@@ -111,51 +114,65 @@ private bool ValidMove(string pressedButton, Vector3 currentTransformPosition)
         {
             case "Up":
                 if (!Physics.Raycast(currentTransformPosition +
-                    new Vector3(0, 0, 1), Vector3.down, out hitInfo, 1, mask))
+                                     new Vector3(0, 0, 1), Vector3.down, out hitInfo, 1, mask))
                 {
                     return false;
                 }
-                return ValidateFloorMove(hitInfo);
+
+                return ValidateFloorMove(hitInfo, Vector3.forward);
 
             case "Down":
                 if (!Physics.Raycast(currentTransformPosition +
-                    new Vector3(0, 0, -1), Vector3.down, out hitInfo, 1, mask))
+                                     new Vector3(0, 0, -1), Vector3.down, out hitInfo, 1, mask))
                 {
                     return false;
                 }
-                return ValidateFloorMove(hitInfo);
+
+                return ValidateFloorMove(hitInfo, Vector3.back);
 
             case "Left":
 
                 if (!Physics.Raycast(currentTransformPosition +
-                    new Vector3(-1, 0, 0), Vector3.down, out hitInfo, 1, mask))
+                                     new Vector3(-1, 0, 0), Vector3.down, out hitInfo, 1, mask))
                 {
                     return false;
                 }
-                return ValidateFloorMove(hitInfo);
+
+
+                return ValidateFloorMove(hitInfo, Vector3.left);
 
             case "Right":
                 if (!Physics.Raycast(currentTransformPosition +
-                    new Vector3(1, 0, 0), Vector3.down, out hitInfo, 1, mask))
+                                     new Vector3(1, 0, 0), Vector3.down, out hitInfo, 1, mask))
                 {
                     return false;
                 }
 
-                return ValidateFloorMove(hitInfo);
+                return ValidateFloorMove(hitInfo, Vector3.right);
         }
 
         return false;
     }
 
-    private bool ValidateFloorMove(RaycastHit hitInfo)
+    private bool ValidateFloorMove(RaycastHit hitInfo, Vector3 direction)
     {
+        if (Physics.Raycast(transform.position, direction, out var hit, 1f)
```

Comment:
If this entire `if` statement can be rewritten to be in one or multiple `bool` variable(s) that would be awesome. If that doesn't work out, leave it.